File: Java/프로그래머스/0/181868. 공백으로 구분하기 2/공백으로 구분하기 2.java
class Solution {
    public String[] solution(String my_string) {
        String[] sp = my_string.split("\\s+");
        int count=0;
        int n=0;
        for(int i=0 ; i<sp.length;i++){
            if(!sp[i].equals("")) count++;
        }
        String answer[]=new String[count];
        for(int i=0; i<sp.length;i++){
            if(!sp[i].equals("")) {
                answer[n]=sp[i];
                n++;
                }
        }
        return answer;
    }
}

File: JavaScript/프로그래머스/0/181868. 공백으로 구분하기 2/공백으로 구분하기 2.js
function solution(my_string) {
    var result = my_string.split(/\s+/);
    var count = 0;
    var n=0;
    for(var i=0; i<result.length; i++){
        if(result[i]!=""){
            count++;
        }
    }
    var answer = new Array(count);
    for(var i=0; i<result.length; i++){
        if(result[i]!=""){
            answer[n]=result[i];
            n++;
        }
    }
    return answer;
}